pip/pip-433.md

```diff
@@ -0,0 +1,172 @@
+
+#### Schemas replication if you have enabled Geo Replication.
+The internal producer of the geo replicator starts with an `auto-produce` schema, copies a new schema it reads from the source cluster to the remote cluster, and it will be stuck once a schema is incompatible to the remote cluster.
+
+# Motivation
+
+#### Issue 1: conflict topic creation if enabled Geo-Replication
+**Steps to reproduce the issue**
+- Configurations of both the source cluster and the remote cluster 
+  - `allowAutoTopicCreation`: `true`
+  - `allowAutoTopicCreationType`: `non-partitioned`
+  - `defaultNumPartitions`: `2`
+- The namespace `public/default` exists, but you have not enabled Geo-Replication for the namespace yet.
+- Start a producer for a topic `persistent://public/default/topic` on the source cluster.
+  - It triggers a partitioned topic with `2` partitions created.
+- Enable namespace-level Geo-Replication for the namespace `public/default`.
+- Start geo replicator for the existing sub-topic `public/default/topic-partition-0`
+  - Without [PIP-414: Enforce topic consistency check](https://github.com/apache/pulsar/pull/24213), the geo replicator will trigger a non-partitioned topic creation, which is named `public/default/topic-partition-0`
+  - With [PIP-414: Enforce topic consistency check](https://github.com/apache/pulsar/pull/24213), the geo replicator get a denied error.
+- However, the user wants to allow the replicator to copy messages to the remote cluster.
+
+#### Issue 2: Replication is stuck because the remote side does not allow schema updates
+**Steps to reproduce the issue**
+- The topic`public/default/topic` has enabled a geo replication.
+- Users controls the topic schema manually, do not allow auto update schema by consumers/producers for both clusters.
+- The internal producer is stuck after the user sets a new schema on the source cluster.
+- However, the user wants to allowthe  replicator to copy the schema to the remote cluster.
+
+# Goals
+
+1. Add an optional choice: always allow the replicator to register schemas if compatible, even if users set `set-is-allow-auto-update-schema` to false.
+2. Add an optional choice: always allow the replicator to trigger a topic auto-creation, including partitioned topics, even if users have disabled topic auto-creation.
+  a. It only triggers auto-creation if the remote side has the same auto-creation policy of partition-type and partition-num.
+3. Checks compatibility between two clusters when enabling namespace-level replication, which includes the following
+  a. All topics’ partitions that have been created should be the same, including `__change_events`
+  b. Auto-creation policies should be the same
+  c. Support for adjusting policies automatically when enabling namespace-level policies, and provide an API to review the changes that will be applied. By the way, this feature will be an option param of enabling namespace-level geo replication
```

Comment:
I think the geo-replication should get rid of the topic auto-creation. So we don't need to sync the policies.

Reply:
> I think the geo-replication should get rid of the topic auto-creation. So we don't need to sync the policies.

This change is used to support the topics that already exist before enabling a namespace-level Geo-Replication. For example:

- Both the source cluster and the remote cluster have a namespace `public/default`
- Source cluster has a system topic `public/default/__change_events`, which is a non-partitioned topic
- Remote cluster has a system topic `public/default/__change_events`, which is a partitioned topic with `2` partitions.
- At the moment, user wants to enable a namespace-level Geo-Replication for `public/defaut`